React/CxxBridge/RCTMessageThread.mm

```diff
@@ -36,7 +36,12 @@
 
 // This is analogous to dispatch_async
 void RCTMessageThread::runAsync(std::function<void()> func) {
-  CFRunLoopPerformBlock(m_cfRunLoop, kCFRunLoopCommonModes, ^{ func(); });
+  CFRunLoopPerformBlock(m_cfRunLoop, kCFRunLoopCommonModes, ^{
+    // Create an autorelease pool each run loop to prevent memory footprint from growing too large, which can lead to performance problems.
+    @autoreleasepool {
+      func();
+    }
+  });
```

Comment:
This will impact pre-TurboModule system as well. The OOM is only when TM is used right? So I think the fix should be done somewhere around the caller of `runAsync` that's specific to TM?

In other words, this code doesn't leak memory pre-TM, why is that? If we can find the logic in the old system that prevents the leak, we should apply the same fix for TM. So this PR might not address the TM-specific reason for the leak.

Reply:
@fkgozali Actually, pre-TurboModule also has the issue, the reason why pre-TurboModule seems not have is it not creates giant autorelease objects, for example, TurboModule create [one `NSInvocation` (autorelease object) for each method call](https://github.com/facebook/react-native/blob/8797a5cfcb8a3dbd233408fb4ac86d78239f5e3f/ReactCommon/turbomodule/core/platform/ios/RCTTurboModule.mm#L474), but pre-TurboModule [only create one by cache](https://github.com/facebook/react-native/blob/aee88b6843cea63d6aa0b5879ad6ef9da4701846/React/Base/RCTModuleMethod.mm#L508). So pre-TurboModule just hidden the issue in @sjchmiela 's demo, but actually it also has if module method create many autorelease objects.

Reply:
> but pre-TurboModule only create one by cache

I see, good catch. I think we should aim to get the caching ideally, but that may be out of scope for this issue. If you're up for some refactoring, that would be helpful :).